tests/test_core.py

```diff
@@ -1,139 +1,98 @@
-from typing import Optional
+from typing import Any, Dict, Optional, SupportsFloat, Tuple
 
 import numpy as np
-import pytest
 
-from gymnasium import core, spaces
-from gymnasium.wrappers import OrderEnforcing, TimeLimit
+from gymnasium import Env, ObservationWrapper, RewardWrapper, Wrapper
+from gymnasium.core import ActionWrapper, ActType, ObsType
+from gymnasium.spaces import Box
+from tests.generic_test_env import GenericTestEnv
 
 
-class ArgumentEnv(core.Env):
```

Comment:
Can you explain these changes? It seems that a bunch of tests were removed and replaced with new ones. Why? I'd be very, **very** careful about removing tests.

Reply:
I should have mentioned this in the top message, add `@abstractmethod` for `Env.step` and `Env.reset` caused a number of issues for testing environments where these functions were not filled out. See the CI in the initial commit. 

To simply the code, I made the environment inherit from `GenericTestEnv` that does have basic implementations of these functions. This simplifies the code and shouldn't have any effect on the tests